### DISCUSSION TITLE
Global cli

### DISCUSSION THREAD

**Discussion by akulamartin** [arminbro/generate-react-cli#130] [Ideas]
Can the cli be made global,lets say like Angular JS  and/or Ember JS?

**Comment by arminbro** [comment]
@akulamartin 

Although generate-react-cli is primarily designed for React, it can be customized to work with other frameworks. By setting up [custom component templates](https://github.com/arminbro/generate-react-cli?tab=readme-ov-file#custom-component-templates) and specifying [custom file](https://github.com/arminbro/generate-react-cli?tab=readme-ov-file#custom-component-files) structures in your **generate-react-cli.json** configuration, you can adapt it for non-react projects. This isn’t an officially supported feature, but it’s possible with these configuration tweaks.

Also, since both Angular and Ember have their own dedicated CLIs, you may find their tools better suited for those frameworks.

Someone recently asked a similar question on  [Twitter](https://x.com/doubleedesign/status/1735760532429512896)  about using generate-react-cli in a Vue project. They were able to successfully integrate it by applying the same setup tweaks I recommended here.

Hope this helps!
